Index: Homework/Task 2.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Homework/Task 2.py b/Homework/Task 2.py
new file mode 100644
--- /dev/null	(date 1663873230431)
+++ b/Homework/Task 2.py	(date 1663873230431)
@@ -0,0 +1,10 @@
+# Выполнил Андрей Кошелев Аналитика | 6 | 3040 | 21.09.2022
+# 2- Напишите программу для. проверки истинности утверждения
+# ¬(X ⋁ Y ⋁ Z) = ¬X ⋀ ¬Y ⋀ ¬Z для всех значений предикат.
+
+for x in range(0, 2):
+    for y in range(0, 2):
+        for z in range(0, 2):
+            comment1 = not(x or y or z)
+            comment2 = not x and not y and not z
+            print(x, y, z, comment1, comment2)
\ No newline at end of file
Index: Homework/Task 3.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Homework/Task 3.py b/Homework/Task 3.py
new file mode 100644
--- /dev/null	(date 1663875002674)
+++ b/Homework/Task 3.py	(date 1663875002674)
@@ -0,0 +1,17 @@
+# Выполнил Андрей Кошелев Аналитика | 6 | 3040 | 21.09.2022
+# 3. Напишите программу, которая по заданному номеру четверти,
+# показывает диапазон возможных координат точек в этой четверти (x и y).
+
+
+
+n = int(input('input quarter number: '))
+if n < 1 or n > 4:
+     print('Please, repeat the input')
+elif n == 1:
+     print('x > 0 and y > 0')
+elif n == 2:
+     print('x < 0 and y > 0')
+elif n == 3:
+     print('x < 0 and y < 0')
+elif n == 4:
+     print('x > 0 and y < 0')
\ No newline at end of file
diff --git a/Homework/Task 4.py b/Homework/Task 4.py
new file mode 100644
diff --git a/Homework/Task 5.py b/Homework/Task 5.py
new file mode 100644
